cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1108,47 +1126,18 @@ private ApplicationDetail enforceApplicationDetailAccess(ApplicationId appId,
   }
 
   // get filter for app specs by artifact name and version. if they are null, it means don't filter.
```

Comment:
I think we'd better make it clear what's the case for `artifactNames`, when it's empty , null.

From the code if it's null, NPE will be thrown.